src/internals/graphql/resolver.ts

```diff
@@ -39,6 +39,8 @@ export type MutationResolver<A, T> = Resolver<
     ? Omit<T, 'query'> & { query: Record<string, never> }
     : T
 >
+export type OverwriteRecord<T, R> = Omit<T, 'record'> & { record: R | null }
```

Comment:
```suggestion
export interface MutationResponse {
  success: boolean,
  query: Record<string, never>
}

export interface MutationResponseWithRecord<R> extends MutationResponse {
  record: R
}
```

This might be easier to use for others.